test/sqs_lambda_to_batch_fargate/lambda/export_logs.py

```diff
@@ -0,0 +1,81 @@
+import json
+import logging
+import os
+import time
+
+import boto3
+
+s3_client = boto3.client("s3")
+s3_resource = boto3.resource("s3")
+logs_client = boto3.client("logs")
+
+logger = logging.getLogger()
+logger.setLevel(logging.INFO)
+
+
+def handler(event, context):
+    logger.info(f"INCOMING EVENT: {event}")
+    if "Batch Job State Change" in event["detail-type"]:
+        if (
+            "FAILED" in event["detail"]["status"]
+            or "SUCCEEDED" in event["detail"]["status"]
+        ):
+            logger.info(f"BUCKET_NAME: {os.environ['BUCKET_NAME']}")
+            try:
+                log_group_name = "/aws/batch/job"
+
+                # Get most recent stream
+                log_streams = logs_client.describe_log_streams(
+                    logGroupName=log_group_name,
+                    orderBy="LastEventTime",
+                    descending=True,
+                    limit=1,
+                )
+
+                # Get log events from the stream
+                log_events = logs_client.get_log_events(
+                    logGroupName=log_group_name,
+                    logStreamName=log_streams["logStreams"][0]["logStreamName"],
+                    startFromHead=True,
+                )
+
+                # Generate a timestamp for the current time
+                timestamp_str = int(time.time())
+                file_name = f"{timestamp_str}.log"
+                with open(f"/tmp/{file_name}", "w") as output_file:
```

Comment:
This will break at some point in the far future in a terrifying and difficult to diagnose fashion. If we are _very lucky_ it will be because the lambda disk is full. More likely, multiple invocations will accidentally get the same `time.time()` value^1. Let's skip the whole thing and use multipart upload rather than upload_file.

1. https://docs.python.org/3/library/time.html says only that `time.time` is "a floating point number", and leaves the details of that to a myriad of system configuration complexities. Then truncating that to an `int` guarantees that any invocations happening in the same second will get the same `/tmp` file, and overwrite each others logs while they're streaming to s3. TL;DR never trust a clock, and in practice let's just multipart upload directly from the stream to s3 & skip /tmp entirely.

Reply:
Done.